rewrite-java/src/main/java/org/openrewrite/java/style/ImportLayoutStyle.java

```diff
@@ -785,14 +785,15 @@ public String toString() {
 
     private static String packageOrOuterClassName(JRightPadded<J.Import> anImport) {
         String typeName = anImport.getElement().getTypeName();
-        if (anImport.getElement().isStatic()) {
+        JavaType.FullyQualified type = TypeUtils.asFullyQualified(anImport.getElement().getQualid().getType());
+        if (anImport.getElement().isStatic() || (type != null && type.getKind() == JavaType.FullyQualified.Kind.Enum)) {
```

Comment:
I don't think the exceptional handling of enums here is quite right; if I alter the above test slightly you can see an import that should be folded, but isn't.

```java
@Issue("https://github.com/openrewrite/rewrite/issues/4165")
@Test
void doNotPreserveOuterEnum() {
    rewriteRun(
      java(
        """
          package com.fasterxml.jackson.core.util;
          
          public enum Spacing { NONE, BEFORE, AFTER, BOTH }
          """,
        SourceSpec::skip
      ),
      java(
        """
          import com.fasterxml.jackson.core.util.DefaultIndenter;
          import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
          import com.fasterxml.jackson.core.util.JacksonFeature;
          import com.fasterxml.jackson.core.util.MinimalPrettyPrinter;
          import com.fasterxml.jackson.core.util.Separators;
          import com.fasterxml.jackson.core.util.Spacing;
          
          class PrettyPrinterTest {
              void print() {
                  System.out.println(Spacing.NONE);
              }
          }
          """,
        """
          import com.fasterxml.jackson.core.util.*;
          
          class PrettyPrinterTest {
              void print() {
                  System.out.println(Spacing.NONE);
              }
          }
          """
      )
    );
}
```

As this fails with:
```diff
diff --git a/PrettyPrinterTest.java b/PrettyPrinterTest.java
index 3267855..c62317b 100644
--- a/PrettyPrinterTest.java
+++ b/PrettyPrinterTest.java
@@ -1,4 +1,5 @@ 
 import com.fasterxml.jackson.core.util.*;
+import com.fasterxml.jackson.core.util.Spacing;
 
 class PrettyPrinterTest {
     void print() {
```

Reply:
Apparently in this case the enum does not get marked as static, checking the flags instead of the kind solves this